### ISSUE TITLE
Upcoming KDE Frameworks 5.32.0

### ISSUE THREAD

**Issue by JenaPlinsky** [psifidotos/Latte-Dock#252]
[Frameworks 5.32.0](https://www.kde.org/announcements/kde-frameworks-5.32.0.php)

**Comment by psifidotos** [comment]
I dont understand...

**Comment by JenaPlinsky** [comment]
I thought it was interesting to read that's all :)

**Comment by psifidotos** [comment]
No prob... We can create a thread for news / announcements here concerning KDE/Plasma/Latte

**Comment by psifidotos** [comment]
Upcoming KDE Frameworks 5.32.0

[Frameworks 5.32.0](https://www.kde.org/announcements/kde-frameworks-5.32.0.php)
This is not an issue, hence you can close or delete

**Comment by psifidotos** [comment]
## Latte Dock for KDE Desktops Is a Slick Plasma Panel Replacement
http://www.omgubuntu.co.uk/2017/03/kde-latte-dock-replace-plasma-panel

**Comment by JenaPlinsky** [comment]
Here are the new personal assistant Mycroft docked in Latte
![spectacle g25867](https://cloud.githubusercontent.com/assets/24593950/24014589/dae06840-0a85-11e7-9249-431c9a89fe0a.png)

**Comment by BigDaddyonTech** [comment]
I just posted a video on this.
Great job by the dev
https://youtu.be/rYoilyd8vCg

**Comment by psifidotos** [comment]
@BigDaddyonTech thank you very much for the beautiful review!!! :)
of coure I am not alone...
@audoban is the other half of the team!!! and we soon hope to be more!!! ;)

**Comment by BigDaddyonTech** [comment]
Well, I have to say that the Dock is absolutely awesome. I thank you both for the efforts you put into it.

**Comment by psifidotos** [comment]
## Latte Dock inside MATE...

@BigDaddyonTech may you are interested, read the comments in the photos for more infos... :)

at the bottom right corner of the screen is the mate panel, I couldnt remove it so I put it there...


![latte-dock-matte1](https://cloud.githubusercontent.com/assets/2216474/24070288/ddc29bd0-0bc2-11e7-8bea-f751b7e7043f.png)

![latte-dock-matte2](https://cloud.githubusercontent.com/assets/2216474/24070293/fb9a342e-0bc2-11e7-8954-8c61d1931393.png)

**Comment by BigDaddyonTech** [comment]
@psifidotos That's awesome!!! I have family members on ubuntu mate. I will have to install mate to test it out for them. Are there any other dependencies that need to be installed other than installing it from the master?

**Comment by psifidotos** [comment]
1. @BigDaddyonTech of course alternative desktop environments are not the focus group at this stage... it is just to demonstrate the capabilities...

2. no more dependencies are needed...

3. I prepared you a branch called **altdesktops** that contains the patch in order to work ok...

**Comment by BigDaddyonTech** [comment]
@psifidotos thanks man. I appreciate it. I am working overtime this week but I will try to test it out asap

**Comment by audoban** [comment]
**Nitrux OS is using Latte by default** https://nxos.org

**Comment by audoban** [comment]
He arrived a little late but I finally finished exams at the University.

@BigDaddyonTech Thank you very much for the review, is awesome 👍

**Comment by psifidotos** [comment]
@BigDaddyonTech some infos and help for alternative desktops.

how to run latte:
1. you can either use the branch **altdesktops**

or

2. you can use the master branch but for your docks you should enable: **Latte Settings -> Tweaks -> Behave as a normal dock window**

if you want to test more plasma widgets than the clock:

- Latte does not provide "Add Widgets" choice and relys on plasma one. So what you could do is to add in Latte your widgets you want from your plasma session and afterwards log in to the alternative desktop to test, e.g. MATE etc... if you have enabled autostart choice, Latte should even autostart when you login in the different desktop environment..

**Comment by psifidotos** [comment]
:) Enjoy moving your launchers!!!!

commit: https://github.com/psifidotos/Latte-Dock/commit/21f36dbc9f3ba0ff2234355431a00e70ba68c481

**Comment by JenaPlinsky** [comment]
Great !!! I like it

While I was testing, the movement of launchers, is appeared this icon ....
![screenshot_20170324_231503](https://cloud.githubusercontent.com/assets/24593950/24315613/42815942-10e8-11e7-88a2-356090f78a1b.png)
![spectacle qq9317](https://cloud.githubusercontent.com/assets/24593950/24315645/796e4b22-10e8-11e7-9260-8d270ddfc058.png)
![spectacle xv9317](https://cloud.githubusercontent.com/assets/24593950/24315670/a8ab075e-10e8-11e7-9789-944582e7d4a2.png)

**Comment by psifidotos** [comment]
@JenaPlinsky I will try to keep the issues clean based on the subject... so I  am going to delete the previous post and please open a dedicated issue for this, I hope you are able to reproduce...

**Comment by psifidotos** [comment]
I cleaned up the issues and I have added also in the wike a page called **_"Tips & Tricks"_**:
https://github.com/psifidotos/Latte-Dock/wiki/Tips-&-Tricks

from my perspective it is good to keep the issues clean especially from red signed issues "bugs".. The comparison between open issues and closed ones is also a good metric for the matureness of a project...

**Comment by psifidotos** [comment]
### Latte first Beta version is out !!! 0.5.98

**Comment by soundprizm** [comment]
First off, this dock is awesome; thank you for all the hard work! The updates and fixes of the 0.5.98 release are great! Have you guys thought about adding an "alternate background" option (e.g., allowing Latte Dock to use its own background image when the plasma taskbar background is transparent)?

**Comment by psifidotos** [comment]
We provide a solid background option at Appearance page for cases that the user doesnot like the theme transparency

**Comment by psifidotos** [comment]
## Latte 0.6 released!!!

http://psifidotos.blogspot.nl/2017/04/latte-dock-v06-fresh-air.html?m=1

**Comment by Petross404** [comment]
Great news! We are looking forward for more awesomeness  👍

**Comment by Petross404** [comment]
[Latte](http://www.phoronix.com/scan.php?page=news_item&px=Latte-Dock-0.6-Released) is becoming viral!! 👍

**Comment by mstraube** [comment]
Latte is in the official Arch Linux repos now and can be installed with `pacman -S latte-dock`.

**Comment by psifidotos** [comment]
> Latte is in the official Arch Linux repos now and can be installed with pacman -S latte-dock

thank u very much for the support!!!!

**Comment by psifidotos** [comment]
## Latte 0.6 in Plasma 5.9 wayland and stock openSUSE Tumbleweed packages!!! It shows just fine probably by using xwayland!!! Only problem is that visibility modes do not work, which is logical... but no crashes in our regard!! :) and mention also that we havent added even one line of wayland support!! :)

![wayland-session](https://cloud.githubusercontent.com/assets/2216474/24677245/c9f422de-198e-11e7-802a-8e489484556c.png)

**Comment by psifidotos** [comment]
playing around a bit with our code... wayland painting for Latte 0.6 :) ... nothing stable of course, no positioning for our windows and no visibility modes, crashing when opening config window... but either way here it is... Latte in wayland....

![image](https://cloud.githubusercontent.com/assets/2216474/24711791/a6082e10-1a29-11e7-94d5-9f5cc920b478.png)

**Comment by psifidotos** [comment]
Help Needed: If you know any developer for #381 in order to provide an extension for thunderbird or another mail manager to send through dbus the number of unread messages.... Then in our regard we will expose these functionality!!!! :)

**Comment by psifidotos** [comment]
:) for news and updates concerning our wayland support you should follow us at: #73 

a new and important milestone in there... :)

**Comment by psifidotos** [comment]
new uniform layout for badges... ;)

![b2edecac-2e59-11e7-841a-520a4a112c11](https://cloud.githubusercontent.com/assets/2216474/25574846/73e542fc-2e5a-11e7-941f-a7b96ec21590.png)

**Comment by JenaPlinsky** [comment]
Cool !!

**Comment by psifidotos** [comment]
playing with inner shadows for the panels instead of external ones... watch out the top dock...

![image](https://cloud.githubusercontent.com/assets/2216474/25823575/2e05cdb0-3445-11e7-8068-59e79a1712aa.png)

![image](https://cloud.githubusercontent.com/assets/2216474/25823803/debc6a24-3445-11e7-837d-e46c346f313b.png)

**Comment by psifidotos** [comment]
**and even without inner shadows!!! Look at the top dock!!! :)**

![image](https://cloud.githubusercontent.com/assets/2216474/25908093/f619d5fe-35b1-11e7-8359-00b968529b34.png)

**Comment by psifidotos** [comment]
I have updated in master the name for the main app metadata file... If you run the install.sh script it will automatically remove the old file... If you want to remove it by yourselves it is named:

`/usr/share/applications/latte-dock.desktop`

the new file is named:
`/usr/share/applications/org.kde.latte-dock.desktop`

**Comment by psifidotos** [comment]
ok... I have extracted the translation strings from the current master as a first step... once a month is good thing I think in order to be more prepared for the releases (the translators to be able to start earlier their work)

**Comment by psifidotos** [comment]
separators everywhere!!! :) look nice even when two of them overlap and of course the parabolic effect works fine!

![image](https://cloud.githubusercontent.com/assets/2216474/26452850/2ca935e2-416a-11e7-97f0-069281b7bb76.png)

**Comment by audoban** [comment]
@psifidotos looks good the separators :D

**Comment by psifidotos** [comment]
> looks good the separators :D

@audoban it was very difficult to provide propery parabolic effect with separators and also to have separators both as applets and launchers... or even separators that overlap, for example when changing activitis and the internal separator is at the same place with an applet separator.... I redesigned all the parabolic effect mechanism... :) but now we have a very robust experience and parabolic effect is not influenced by separators and hidden applets even if these are at the row.... ;)

**Comment by audoban** [comment]
@psifidotos Very nice work, and the parabolic effect also feels natural.

**Comment by psifidotos** [comment]
**Three important news:**

1. If someone has a multi-screen environment, please test #522 and tell us if it works as expected...
2. I have uploaded the monthly publishing for translation strings... If you are a translator you can update your language for this month!
3. some wayland code has already landed in our master branch so you should be able to open your Latte in wayland and the docks should be positioned correctly... 
3.1 **Notice 1:** Latte crashes on every exit but there is a big chance that this isnt our fault, bug: (https://bugs.kde.org/show_bug.cgi?id=380828)
3.2 **Notice 2:** The Dodge visibility modes havent implemented yet... so Dodge modes and Always Visible are not working as expected... **_"Windows Go Below"_** and **_"Auto-Hide"_** should work ok in wayland...

**Comment by psifidotos** [comment]
What is coming... :)

multiple layouts that the user can even assign them to activities to automaticly change into them... ;)

![image](https://user-images.githubusercontent.com/2216474/28285906-108e6d10-6b3f-11e7-8121-fd974f8b0e62.png)

**Comment by psifidotos** [comment]
Layouts Manager... its view is fully working... :) only thing remains is to update the underground layout settings... :) hehe

![manager1](https://user-images.githubusercontent.com/2216474/28379581-dad22f5c-6cbc-11e7-8206-044d02d71a80.png)

![manager2](https://user-images.githubusercontent.com/2216474/28379587-dd9fac50-6cbc-11e7-8a48-4783e0c58558.png)

**Comment by JenaPlinsky** [comment]
Absolutely amazing !!!!!

**Comment by rikmills** [comment]
latte-dock 0.6.2 is now available in the kubuntu backports ppa for 16.04 and 17.04

https://launchpad.net/~kubuntu-ppa/+archive/ubuntu/backports/+packages?field.name_filter=latte-dock

**Comment by psifidotos** [comment]
Thunderbird plugin by Jeff Hodd for our master version, upcoming 0.7 ;)

https://sourceforge.net/projects/bluestarlinux/files/extra/lattedock-thunderbird.tar.gz

extract it and inside you can find the .xpi file...

**Comment by psifidotos** [comment]
**IMPORTANT NOTICE ::::** In the next days after some days of review in master we will merge the new layouts infrastructure which provides a new configuration system!!! **So please take backups!!!!** The transition should be smooth as we provide many automatic and manual importing mechanisms for old exported files!!! If everything goes well you shouldnt notice any difference except the new, improved and powerfull functionality for dynamic layouts!!!!

**Comment by psifidotos** [comment]
From the Layouts land!!! An info window now is appearing whenever your Layout is switched automatic because of a current activity change.
**TRANSLATORS:** I have merge the translation strings and tested it with my native language to be sure that it works correctly. If you are a translator now would be a good time to translate. I believe there will be one more translation strings merge in 10-14 days and after that we go for v0.7 release... have fun.... 

![image](https://user-images.githubusercontent.com/2216474/28573527-9faa3532-7153-11e7-8c1e-eadfc3ab9c6b.png)

**Comment by psifidotos** [comment]
Latte plasmoid now provides an Alternative option when Latte is in editing mode in order to switch to plasma taskmanagers. All heuristic and smart algorithms of Latte for now on take into account also the plasma taskmanagers.

e.g. when Latte loads a layout checks if there will be a tasks plasmoid present. If it doesnt then forcefully loads the oldest dock containing a tasks plasmoid which is on an explicit screen that is not present now.

![image](https://user-images.githubusercontent.com/2216474/28722323-011bfff2-73bc-11e7-9370-b6f33c5f8e2b.png)

**Comment by psifidotos** [comment]
Ok.... I think we can have a feature freeze ... We have got around two weeks for releasing the incredible 0.7 version!!! :) so this is a call for translators also!!! We have one week to improve the translation strings in case translators want any improvement and one more week for sending the final translations!!! ;)

have fun everyone!!!

**Comment by psifidotos** [comment]
Different Layouts per Activities... :) enjoy...

https://www.youtube.com/watch?v=kIsX7LSLeik

**Comment by psifidotos** [comment]
**_We entered beta phase with version 0.6.98. ... around 10 days before releasing v0.7 for bug fixes and translations..._**

**Comment by psifidotos** [comment]
**_Ok... Latte Dock v0.7 is here!!! :) enjoy everyone..._**

https://psifidotos.blogspot.nl/2017/08/latte-dock-v07-tornado-is-coming.html

**Comment by rikmills** [comment]
@psifidotos Congratulations!

**Comment by psifidotos** [comment]
@rikmills have you built also 0.7 packages for Neon which uses ubuntu 16.04? There are some users that asked me

**Comment by rikmills** [comment]
@psifidotos Yes, I did a test build against Neon's user edition repo. Can't do that in a ubuntu ppa, so for now I've just put the result here: https://github.com/rikmills/debs

**Comment by psifidotos** [comment]
perfect!!

**Comment by psifidotos** [comment]
because of its importance, I added a "New" button in Layouts manager... I hope you can manage to translate it before the 0.7.1 release which is going to be the first week of September...

**Comment by psifidotos** [comment]
![image](https://user-images.githubusercontent.com/2216474/29750541-1e8e1d9a-8b4a-11e7-9286-d85f92867c36.png)

**Comment by rikmills** [comment]
Looks hopeful on a move to KDE infra :)

https://cgit.kde.org/sysadmin/repo-metadata.git/commit/?id=73ddc6bf1362140e14b0d7e6a083c9b6e1e21178

**Comment by psifidotos** [comment]
> Looks hopeful on a move to KDE infra :)

yes... both developers we decided that this is the best for Latte...

**https://psifidotos.blogspot.com/2017/08/latte-bug-fix-release-v071.html#more**

**Comment by rikmills** [comment]
@psifidotos Great. This should make it easier to get latte-dock into KDE Neon, Ubuntu and Debian

**Comment by psifidotos** [comment]
Well everyone!!! From tomorrow all the Latte translations should  be done through https://l10n.kde.org/ and the relevant localization teams!! all the current translations will be merged also! so in the future if you want to continue tranlating Latte and improve its internationalization presence please join a kde localization team and support the project from there!!! :)

**Comment by omni6** [comment]
> so in the future if you want to continue translating Latte and improve its internationalization presence please join a kde localization team and support the project from there!!! :)

done
now the easy times are gone

**Comment by psifidotos** [comment]
ok.... :) we are getting ready for the 0.7.2 version that fixes many things!!! I have updated the master and v0.7 branch in order to be on par with the kde repo... so you can test things out ... as a notice... from github you will not be able to install translations unless you use:

`./install.sh --translations`

it downloads translations from kde repos but it takes some time to finish...

**Comment by rikmills** [comment]
@psifidotos Congrats on 0.7.2 release! :)

**Comment by JenaPlinsky** [comment]
Happy Christmas and best wishes for a fantastic happy and serene 2018 to all of you

**Comment by psifidotos** [comment]
Latte Layouts online ;)  : https://store.kde.org/browse/cat/417/ord/latest/

![](https://user-images.githubusercontent.com/2216474/34541944-f6b61fa8-f0e2-11e7-9ef9-47073ddc504a.png)

**Comment by psifidotos** [comment]
## Latte Dock v0.7.3

http://psifidotos.blogspot.nl/2018/01/latte-bug-fix-release-v073-and-some-news.html

**Comment by psifidotos** [comment]
What do you think if the user was able to choose a local file for the editing mode background per layout?

![redvelvet](https://user-images.githubusercontent.com/2216474/36219457-26337eb2-11c0-11e8-9bb8-1bb0058cc857.png)

![cartoonish](https://user-images.githubusercontent.com/2216474/36219563-8565f7f2-11c0-11e8-9fb1-548ffba959af.png)

**Comment by psifidotos** [comment]
![default](https://user-images.githubusercontent.com/2216474/36343918-4d3dabc4-141b-11e8-9062-1d401d264bbb.png)

**Comment by JenaPlinsky** [comment]
Very nice

**Comment by rikmills** [comment]
@psifidotos Do you have an ETA for a new tarball release from master branch?

**Comment by psifidotos** [comment]
> Do you have an ETA for a new tarball release from master branch?

what is ETA ?

**Comment by rikmills** [comment]
**E**stimated **T**ime of **A**rrival

Very common usage, but guess not from your locale. Sorry.

**Comment by psifidotos** [comment]
> Estimated Time of Arrival

maybe this summer... after 18.04 because I want a better wayland experience...

**Comment by rikmills** [comment]
Ok. Kool. Thanks.

**Comment by psifidotos** [comment]
reworked the editing mode patters... ;) more contrast and naturality....

![default](https://user-images.githubusercontent.com/2216474/36802956-f58c0fe0-1cbe-11e8-9c8c-a8e2908776d8.png)

**Comment by psifidotos** [comment]
new splitters following the plasma theme... ;)

![default](https://user-images.githubusercontent.com/2216474/36944674-eb4b0aca-1fa9-11e8-84e1-d693c4b778ac.png)

**Comment by psifidotos** [comment]
Lets welcome KWin Edges support in our master version.... ;)
https://github.com/psifidotos/Latte-Dock/commit/1f0acbc3b587476181b12abd2a706da942e1cadd

![kwin_edges2](https://user-images.githubusercontent.com/2216474/38161275-ca7be022-34d4-11e8-8bff-a0eb7689202b.png)
![kwin_edges3](https://user-images.githubusercontent.com/2216474/38161292-05b18a70-34d5-11e8-856c-2d717b23eacd.png)

**Comment by psifidotos** [comment]
Add your internal separators in your tasks through drag n' drop. Drag the separator applet from the plasma Widgets Expoler and when you  drop it above a task in the taskmanager Latte automagically adds it as internal separator... ;) 
7ce5938312ed8634f877bc1e8b7e0db9a055733c
